### PR TITLE
📖 Remove outdated capi book roadmap

### PR DESCRIPTION
**Pull request by oscr**
Removes the outdated roadmap information from the CAPI book as discussed #5519. Since there is no replacement and to avoid leaving a blank page I took the liberty to just refer the reader to GitHub discussions. Not a perfect solution, but better than having a blank page or outdated information in the book.

### CODE REVIEW

File: docs/book/src/roadmap.md [hunk 1]
```diff
@@ -1,52 +1,3 @@
...
-> and can be moved up into a milestone if there is enough interest.
-
-|Area|Description|Issue/Proposal|
-|--|--|--|
-|Security|Machine attestation for secure kubelet registration|[#3762](https://github.com/kubernetes-sigs/cluster-api/issues/3762)|
-|Conformance| Define Cluster API provider conformance|TBA|
-|Core Improvements|Pluggable MachineDeployment upgrade strategies|[#1754](https://github.com/kubernetes-sigs/cluster-api/issues/1754)|
-|UX|Simplified cluster creation experience|[#1227](https://github.com/kubernetes-sigs/cluster-api/issues/1227)|
-|Bootstrap, Infrastructure|Document approaches for infrastructure providers to consider for securing sensitive bootstrap data|[#1739](https://github.com/kubernetes-sigs/cluster-api/issues/1739)|
+The roadmap is a constant work in progress, subject to frequent revision. You can find it at [GitHub](https://github.com/kubernetes-sigs/cluster-api/discussions). Please feel free to participate in the discussion!
```
Comment (killianmuldoon) [inline docs/book/src/roadmap.md:3]: Might be better to link directly to the specific discussion

```suggestion
You can find the Cluster API roadmap discussion at [GitHub](https://github.com/kubernetes-sigs/cluster-api/discussions/5556). Please feel free to participate!
```

Comment (oscr) [reply docs/book/src/roadmap.md]: Sounds like a solid suggestion. Thanks!

**Review by killianmuldoon — APPROVED** [review]
/lgtm

Thanks for this - removing this outdated roadmap is a big improvement to the docs at this point.

**Comment by oscr** [comment]
I have squashed the commits, as was requested in #6666.

**Comment by sbueringer** [comment]
Thx!
/lgtm

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**